src/hooks/useAbortController.ts

```diff
@@ -0,0 +1,17 @@
+import { useCallback, useRef } from 'react'
+
+export function useAbortController() {
+  const abortControllerRef = useRef<AbortController>()
+
+  const abortControllerSignal = useCallback(() => {
+    if (abortControllerRef.current) {
+      abortControllerRef.current.abort()
+    }
+
+    abortControllerRef.current = new AbortController()
+
+    return abortControllerRef.current.signal
+  }, [])
+
+  return { abortControllerSignal }
+}
```

Comment:
Good job @berteotti . Suggest some minor variable name changes. If you feel its ok you can ignore.

```suggestion
export function useAbortController() {
  const abortControllerRef = useRef<AbortController>()

  const getAbortSignal = useCallback(() => {
    if (abortControllerRef.current) {
      abortControllerRef.current.abort()
    }
    abortControllerRef.current = new AbortController()
    return abortControllerRef.current.signal
  }, [])

  return getAbortSignal
}
```